src/clr-angular/data/datagrid/render/row-renderer.ts

```diff
@@ -3,46 +3,32 @@
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
-import { AfterContentInit, ContentChildren, Directive, OnDestroy, QueryList } from '@angular/core';
-import { Subscription } from 'rxjs';
-
-import { DatagridRenderStep } from '../enums/render-step.enum';
+import { AfterViewInit, ContentChildren, Directive, QueryList } from '@angular/core';
 
 import { DatagridCellRenderer } from './cell-renderer';
-import { DatagridRenderOrganizer } from './render-organizer';
+import { ColumnsService } from '../providers/columns.service';
 
 @Directive({ selector: 'clr-dg-row, clr-dg-row-detail' })
-export class DatagridRowRenderer implements AfterContentInit, OnDestroy {
-  constructor(private organizer: DatagridRenderOrganizer) {
-    this.subscriptions.push(
-      organizer.filterRenderSteps(DatagridRenderStep.ALIGN_COLUMNS).subscribe(() => this.setWidths())
-    );
-  }
-
-  private subscriptions: Subscription[] = [];
-  ngOnDestroy() {
-    this.subscriptions.forEach(sub => sub.unsubscribe());
-  }
+export class DatagridRowRenderer implements AfterViewInit {
+  @ContentChildren(DatagridCellRenderer) private cells: QueryList<DatagridCellRenderer>;
 
-  @ContentChildren(DatagridCellRenderer) cells: QueryList<DatagridCellRenderer>;
+  constructor(private columnsService: ColumnsService) {}
 
-  private setWidths() {
-    if (this.organizer.widths.length !== this.cells.length) {
```

Comment:
Have you checked it with dynamic columns? I think we need this guard. This will make datagrid widths broken in the case dynamics columns.

Reply:
Yes I've tested with `*ngIf` on columns and toggling them. We reset the state when a column is removed and reattached, which is true before this change as well. It does calculate an initial strict width if it was set on the column like `[style.width.px]="100"`, otherwise reiszed widths are cleared.

Reply:
Sounds good. Thanks for confirming.